bin/comment_on_pr

```diff
@@ -1,14 +1,141 @@
 #!/bin/bash -eu
 
-PR_MESSAGE_FILE=$1
-GITHUB_TOKEN=$2
+# This script is used to comment on a pull request, and must be run in a buldkite PR step.
+#
+# Usage:
+#    [GITHUB_TOKEN=<token>] comment_on_pr [--id <comment-id>] [--if-exist update|delete] [--github-token <token>] <comment>
+#
+# Examples:
+#
+#    # Post a new comment which can be uniqly identifiered by the given comment id,
+#    # or update the existing comment that's associated with the given comment id.
+#    # You can choose to delete the existing comment if it exists by passing `--if-exist delete`.
+#    comment_on_pr --id comment-id [--if-exist update|delete] "This is a comment"
+#
+#    # Delete the comment associated with the given comment id
+#    comment_on_pr --id comment-id [--if-exist delete]
+#
+#    # Deprecated: Post a new comment to the PR
+#    comment_on_pr "This is a comment" <github-token>
+#
+# Please note the comment argument can be a path to a file containing the comment, or a markdown.
 
-if ! command -v gh &> /dev/null; then
-	brew install gh
+# Check dependencies and print their versions for diagnosis purposes
+cat <<EOF
+> jq --version
+$(jq --version)
+
+> curl --version
+$(curl --version)
+
+EOF
+
+if [[ "${BUILDKITE_PULL_REQUEST:-invalid}" == "invalid" ]]; then
+    echo "Error: this tool can only be called from a Buildkite PR job"
+    exit 1
+fi
+
+# Default options
+opt_comment_id=""
+opt_if_exist="update"
+opt_github_token="${GITHUB_TOKEN-:}"
+
+# If there are two arguments and they don't start with an option, then we assume
+# this utility is called in the deprecated way: comment_on_pr <comment> <github-token>
+if [[ $# == 2 && "$1" != "--" ]]; then
```

Comment:
🙈 Addressed in https://github.com/Automattic/a8c-ci-toolkit-buildkite-plugin/pull/51/commits/d9fed6260cb02315ea076116980aa9f1a6e34f6d

> if that's a requirement we have and agree to keep

Having the options before the argument is an intention "feature". The only reason is it's easy and straight-forward to parse such cli arguments. This is not something that I explicitly brought up for discussion, because I thought it would be a nice and simple improvement to the existing way of parsing arguments which is based on the argument position. We can definitely standardize the argument parsing by using `getopt`, but I think we could do it in a separate PR if it's needed?

Reply:
To be clear, I don't mind if we have the requirement that all `--` options have to appear first, especially if it makes the implementation simpler; given we're the users of those utilities and all, it probably isn't worth it to make them _that_ flexible around the parameters order, and I'm ok to keep the current parsing logic that expects `--` first.

I was just mentioning this in order to make sure that your test on `if [[ $# == 2 && "$1" != "--" ]]; then` didn't make any incorrect assumptions or failed to consider a case that we were supposed to support (arbitrary param order), but I'm ok with us not supporting that case in the first place.